extensions/warp-ipfs/src/store/message/community_task.rs

```diff
@@ -0,0 +1,3142 @@
+            }
+            CommunityTaskCommand::GrantCommunityChannelPermission {
+                response,
+                channel_id,
+                permission,
+                role_id,
+            } => {
+                let result = self
+                    .grant_community_channel_permission(channel_id, permission, role_id)
+                    .await;
+                let _ = response.send(result);
+            }
+            CommunityTaskCommand::RevokeCommunityChannelPermission {
+                response,
+                channel_id,
+                permission,
+                role_id,
+            } => {
+                let result = self
+                    .revoke_community_channel_permission(channel_id, permission, role_id)
+                    .await;
+                let _ = response.send(result);
+            }
+            CommunityTaskCommand::GrantCommunityChannelPermissionForAll {
+                response,
+                channel_id,
+                permission,
+            } => {
+                let result = self
+                    .grant_community_channel_permission_for_all(channel_id, permission)
+                    .await;
+                let _ = response.send(result);
+            }
+            CommunityTaskCommand::RevokeCommunityChannelPermissionForAll {
+                response,
+                channel_id,
+                permission,
+            } => {
+                let result = self
+                    .revoke_community_channel_permission_for_all(channel_id, permission)
+                    .await;
+                let _ = response.send(result);
+            }
+            CommunityTaskCommand::SendCommunityChannelMessage {
+                response,
+                channel_id,
+                message,
+            } => {
+                let result = self
+                    .send_community_channel_message(channel_id, message)
+                    .await;
+                let _ = response.send(result);
+            }
+            CommunityTaskCommand::DeleteCommunityChannelMessage {
+                response,
+                channel_id,
+                message_id,
+            } => {
+                let result = self
+                    .delete_community_channel_message(channel_id, message_id)
+                    .await;
+                let _ = response.send(result);
+            }
+            CommunityTaskCommand::EventHandler { response } => {
+                let sender = self.event_broadcast.clone();
+                let _ = response.send(sender);
+            }
+            CommunityTaskCommand::Delete { response } => {
+                let result = self.delete().await;
+                let _ = response.send(result);
+            }
+        }
+    }
+}
+
+impl CommunityTask {
+    pub async fn delete(&mut self) -> Result<(), Error> {
+        // TODO: Maybe announce to network of the local node removal here
+        //self.document.messages.take();
```

Comment:
any reason why this would be commented out?

Reply:
`self.document.messages.take();` is commented because it doesn't exist yet. Messaging isn't implemented yet, but when it is, it won't be on the root level of the community, it will be on the individual channels. I commented it so it would compile but also so I wouldn't forget to do it in my next PR. Added a comment here to make it more clear. Thanks 👍